pkg/tests/tasks/security/certificate/cert_manager_test.go

```diff
@@ -53,32 +58,63 @@ func TestCertManager(t *testing.T) {
 		helm.Namespace(meshNamespace).
 			Chart("jetstack/cert-manager-istio-csr").
 			Release("istio-csr").
+			Version("v0.6.0").
 			ValuesString(istioCsrValues(meshNamespace, smcpName)).
 			Install(t)
 		oc.WaitPodsReady(t, meshNamespace, "app=cert-manager-istio-csr")
 
-		t.LogStep("Deploy the cert-manager in SMCP")
-		oc.ApplyString(t, meshNamespace, createSMCPWithCertManager(smcpName, meshNamespace, ns))
-		oc.WaitSMCPReady(t, meshNamespace, smcpName)
-
-		t.LogStep("Install httpbin and sleep")
-		app.InstallAndWaitReady(t, app.Httpbin(ns), app.Sleep(ns))
-
-		t.LogStep("Check if httpbin returns 200 OK ")
-		retry.UntilSuccess(t, func(t test.TestHelper) {
-			oc.Exec(t,
-				pod.MatchingSelector("app=sleep", ns),
-				"sleep",
-				`curl http://httpbin:8000/ip -s -o /dev/null -w "%{http_code}"`,
-				assert.OutputContains(
-					"200",
-					"Got expected 200 OK from httpbin",
-					"Expected 200 OK from httpbin, but got a different HTTP code"))
-		})
+		for _, ver := range []string{"v2.3", "v2.4"} {
+			t.NewSubTest(ver).Run(func(t test.TestHelper) {
+				t.Cleanup(func() {
+					app.Uninstall(t, app.Httpbin(ns.Foo), app.Sleep(ns.Foo))
+				})
+
+				t.LogStep("Deploy SMCP " + ver)
+				oc.ApplyString(t, meshNamespace, createSMCPWithCertManager(smcpName, meshNamespace, ns.Foo, ver))
+				oc.WaitSMCPReady(t, meshNamespace, smcpName)
+
+				t.LogStep("Verify that istio-ca-root-cert created in proper namespaces")
+				retryOpts := retry.Options().MaxAttempts(10).DelayBetweenAttempts(1 * time.Second)
+				retry.UntilSuccessWithOptions(t, retryOpts, func(t test.TestHelper) {
+					oc.LogsFromPods(t, meshNamespace, "app=cert-manager-istio-csr",
```

Comment:
Why not use the existing `oc.Logs` function? It does virtually the same thing as the `LogsFromPods` function. You can use the `pod.MatchinSelector()` pod locator function to specify the selector. You do have to specify the container name, though, but if you can, you could modify the function so that using an empty string for the container name causes the function to either use the single container (in single-container pods) or use `--all-containers`.

Reply:
I deploy 2 replicas of istio-csr, and `oc.Logs` requires `PodLocator` that returns an error when more than 1 pod match a given selector, so I had to implement another function.

Reply:
Ah, ok... it would still be great if we could consolidate the two functions into one, but only if it's possible without making the original function harder to use (e.g. by introducing an additional non-optional parameter).

Reply:
I wanted to do this, but it would result in a more complicated function, because the existing one relies on `PodLocatorFunc` that returns `NamespacedName` for a single pod, so I don't have good idea how to easily consolidate them.

Reply:
We should probably change the PodLocatorFunc to return a list of pods.. and then have the consumer of that function (like oc.Exec or oc.Logs) throw the error if it can only work with a single pod. 

But I wonder why you even deploy two instances of istio-csr instead of just one. Does the test require two?

Reply:
I deploy 2 instances, because istio-csr performs leader election and only one instance at the time watches namespaces and creates config maps. It would work with one instance as well, but I thought that it may be better to test istio-csr in a configuration that customers will use on production to anticipate potential problems.